core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -521,8 +521,10 @@ public void channelzMembership_subchannel() throws Exception {
     MockClientTransportInfo transportInfo = transports.poll();
     assertNotNull(transportInfo);
     assertTrue(channelz.containsClientSocket(transportInfo.transport.getLogId()));
+    transportInfo.listener.transportReady();
```

Comment:
Adding this to this test looks fine/natural (so no need for a change). I will just note that it shouldn't be _necessary_ for the transport to become ready before being shutdown. That happens during negotiation failure, for example.

Reply:
this is also related to reconnect, the shutdown triggers backoff and reconnect so the subchannel is not removed from channelz which makes the test fail.